packages/jest-core/src/SearchSource.ts

```diff
@@ -113,27 +111,27 @@ export default class SearchSource {
       total: allPaths.length,
     };
 
-    const testCases = Object.assign({}, this._testPathCases);
+    const testCases = this._testPathCases.slice(0);
     if (testPathPattern) {
       const regex = testPathPatternToRegExp(testPathPattern);
-      (testCases as TestPathCasesWithPathPattern).testPathPattern = (
-        path: Config.Path,
-      ) => regex.test(path);
-      (data.stats as TestPathCaseWithPathPatternStats).testPathPattern = 0;
+      testCases.push({
+        isMatch: (path: Config.Path) => regex.test(path),
+        stat: 'testPathPattern',
+      });
+      data.stats.testPathPattern = 0;
```

Comment:
how about making `testPathPattern` unconditinal like other stats and define it in https://github.com/facebook/jest/pull/8184/files#diff-db8ff11af2e1c67c6847136353c98962R108? This way we could adjust the Stats typedef and avoid overriding types in https://github.com/facebook/jest/pull/8184/files#diff-db8ff11af2e1c67c6847136353c98962R128

Reply:
With PRs that are meant to be optimizations, I try to not change any behavior. I don't know into what other parts of the system that would leak. If it causes a bug, I'd rather not have the whole PR reverted.

Happy to change it to be always-present, but would prefer in a different PR.

Reply:
Fair enough